### PR TITLE
Return empty responses for unsupported features

### PR DESCRIPTION
**Pull request by olafurpg**
Instead of crashing with exception.

### CODE REVIEW

**Comment by olafurpg** [comment]
`null` is a valid response for all of those requests according to the spec.

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala [hunk 1]
```diff
@@ -451,6 +451,110 @@ class MetalsLanguageServer(
...
+      scribe.warn("Rename symbol is not supported.")
+      null
+    }
+
+  @JsonRequest("textDocument/references")
+  def references(
+      position: ReferenceParams
+  ): CompletableFuture[util.List[Location]] =
+    CompletableFutures.computeAsync { _ =>
+      scribe.warn("Find references is not supported.")
```
Comment (gabro) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala]: Super minor but I'd use the request identifier in the warning message "textDocument/references is not implemented"

Comment (laughedelic) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala]: I support this. "textDocument/references is not implemented **yet**" 😄

Comment (olafurpg) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala]: I agree with including the method name, I'm on the fence to promise what will get implemented.

Comment (laughedelic) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala]: I'm just joking 😉

**Comment by olafurpg** [comment]
Appveyor CI is a bit overloaded this morning, just gonna merge since this is an innocent change

**Comment by gabro** [comment]
👍 it's a bit disheartening that these clients don't even respect declared capabilities properly, but it's a safe change to make.

**Comment by olafurpg** [comment]
My sympathy goes to the heroes writing language clients in vim script 🙏

**Pull request merged**